zjb/db.py

```diff
@@ -74,5 +80,32 @@ def create_or_update(cls, session, project, branch, pipeline, job,
         session.commit()
         return instance
 
+    @classmethod
+    def update_note(cls, session, project, branch, pipeline, job, note):
+        instance = session.query(cls).filter_by(project=project,
+                                                branch=branch,
+                                                pipeline=pipeline,
+                                                job=job).one_or_none()
+        if instance:
+            instance.note = note
+            instance.note_updated = datetime.datetime.now()
```

Comment:
This line is not necessary, as you added the in-sql handler (`onupdate=func.now()`).
In General, I would drop the `update_node()` from this pull request for now.

Reply:
done